profile/src/hierarchical_deterministic/bip32/hd_path_component.rs

```diff
@@ -0,0 +1,77 @@
+const BIP32_HARDENED: u32 = 2147483648;
+
+pub type HDPathValue = u32;
+
+#[derive(
+    Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, uniffi::Record,
+)]
+pub struct HDPathComponent {
```

Comment:
TODO: migrate to use `uniffi::custom_newtype!(HDPathComponent, HDPathValue);` and let `HDPathComponent` be a newtype instead.